### PR TITLE
Mocks Logging

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
- [x] Ready

## Description
Adds a `run_tests.log` file to the artifacts for each server instance `Run Tests` circleci job, to which `mock_server.py` logs extra debugging statements. The outputs of `stdout` and `stderr` of the `mitmproxy` process are logged there as well.

### CODE REVIEW

**Comment by avidan-H** [comment]
Adding the `docs-approved` label since there is no documentation to review here.

File: Tests/test_content.py [hunk 1]
```diff
@@ -899,6 +899,7 @@ def execute_testing(tests_settings, server_ip, mockable_tests_names, unmockable_
                 updating_mocks_msg = "Pushing new/updated mock files to mock git repo."
                 prints_manager.add_print_job(updating_mocks_msg, print, thread_index)
                 ami.upload_mock_files(build_name, build_number)
+                prints_manager.execute_thread_prints(thread_index)
```
Comment (yaakovi) [inline Tests/test_content.py:902]: why not execute it directly under the finally? why specifically for AMI on the master branch?

File: Tests/mock_server.py [hunk 2]
```diff
@@ -191,6 +191,10 @@ def __init__(self, public_ip,
         self.empty_files = []
         self.rerecorded_tests = []
 
+        self.logs_file = open(os.getenv('TESTS_LOGS'), 'a') if os.getenv('TESTS_LOGS') else open(
+            '/home/circleci/project/artifacts/run_tests.log', 'a')
+        self.print_to_log = lambda *objects: print(*objects, file=self.logs_file)
```
Comment (yaakovi) [inline Tests/mock_server.py:196]: consider also adding `flush=True`

Comment (avidan-H) [reply Tests/mock_server.py:196]: good idea.

File: Tests/test_content.py (hunk 1 above)
Comment (avidan-H) [reply Tests/test_content.py:902]: not sure I understand this comment.

Comment (yaakovi) [reply Tests/test_content.py:902]: why is that in the condition? why not run at any time in the `finally` block

**Pull request merged**